### PR TITLE
Fixes #35919 - Change content source UX improvements

### PR DESCRIPTION
**Pull request by stejskalleos**
#### What are the changes introduced in this pull request?

- Added bread crumb navigation
- When no hosts are found, still show title & bread-crumb nav
- Form button changed to "Update"
- Notification changed to `info` with steps what to do next

ccs

#### Considerations taken when implementing this change?
- https://projects.theforeman.org/issues/35919
- Discussion with UX team

#### What are the testing steps for this pull request?
- Go to hosts
- Select host with content source
- Select Change content source from actions
- Check the Form

### CODE REVIEW

**Comment by stejskalleos** [comment]
Hi @jeremylenz, could you please do the review? It's another follow up UX PR to https://github.com/Katello/katello/pull/10297

File: webpack/scenes/Hosts/ChangeContentSource/components/ContentSourceTemplate.js [hunk 1]
```diff
@@ -39,25 +40,25 @@ const ContentSourceTemplate = ({ template, jobInvocationPath }) => {
...
-        <h1>
-          {__("What's next?")}
-        </h1>
-        <p>
-          {jobInvocationPath && (
-            <>
-              <a href={jobInvocationPath}>
-                {__('Run job invocation')}
-              </a>
+        <Alert variant="info" title="Hosts successfully updated" className="margin-top-20">
```
Comment (jeremylenz) [inline webpack/scenes/Hosts/ChangeContentSource/components/ContentSourceTemplate.js:43]: Title needs to be translated.

File: webpack/scenes/Hosts/ChangeContentSource/components/ContentSourceTemplate.js [hunk 2]
```diff
@@ -39,25 +40,25 @@ const ContentSourceTemplate = ({ template, jobInvocationPath }) => {
...
           &nbsp;
-              {__('to update configuration on all hosts, or')}
-            </>
-          )}
-        &nbsp;
-          {__('update configuration on the hosts manually:')}
-        </p>
+            {__('update configuration on the hosts manually:')}
+          </p>
+        </Alert>
```
Comment (jeremylenz) [inline webpack/scenes/Hosts/ChangeContentSource/components/ContentSourceTemplate.js:57]: To ensure proper translation, please use `<FormattedMessage>` similar to https://github.com/Katello/katello/blob/47492881d123b13c2e7973c4e1db965971d45657/webpack/scenes/Subscriptions/SubscriptionsPage.js#L223-L235

File: webpack/scenes/Hosts/ChangeContentSource/index.js [hunk 3]
```diff
@@ -86,45 +92,68 @@ const ChangeContentSourcePage = () => {
...
+      </Head>
+      <PageSection
+        isFilled
+        variant="light"
+      >
+        <div className="margin-left-20">
+          <BreadcrumbBar
+            breadcrumbItems={[
+              { caption: __('Hosts'), url: foremanUrl('/hosts') },
+              { caption: 'Change host content source' },
```
Comment (jeremylenz) [inline webpack/scenes/Hosts/ChangeContentSource/index.js:108]: When changing content source for a single host, can that host be added to the breadcrumbs here?

File: webpack/scenes/Hosts/ChangeContentSource/index.js [hunk 4]
```diff
@@ -53,6 +56,8 @@ const ChangeContentSourcePage = () => {
   const [environmentId, setEnvironmentId] = useState('');
   const [contentViewId, setContentViewId] = useState('');
 
+  const noSearch = getHostIds(urlParams.host_id).length === 0 && urlParams.searchParam === '';
```
Comment (jeremylenz) [inline webpack/scenes/Hosts/ChangeContentSource/index.js:59]:
```suggestion
  const noHostSpecified = getHostIds(urlParams.host_id).length === 0 && urlParams.searchParam === '';
```

File: webpack/scenes/Hosts/ChangeContentSource/index.js [hunk 5]
```diff
@@ -86,45 +92,68 @@ const ChangeContentSourcePage = () => {
...
+            >
+              {__('Change host content source')}
+            </Title>
+          </GridItem>
+          {noSearch &&
+            <GridItem span={7}>
+              <Alert
+                variant="danger"
+                className="margin-top-20"
+                title={__('No hosts with content source found!')}
```
Comment (jeremylenz) [inline webpack/scenes/Hosts/ChangeContentSource/index.js:126]:
```suggestion
                title={__('No hosts were specified')}
```

File: webpack/scenes/Hosts/ChangeContentSource/index.js [hunk 6]
```diff
@@ -86,45 +92,68 @@ const ChangeContentSourcePage = () => {
...
+              environments={environments}
+              handleEnvironment={handleEnvironment}
+              environmentId={environmentId}
+              contentViews={contentViews}
+              handleContentView={setContentViewId}
+              contentViewId={contentViewId}
+              contentSources={contentSources}
+              contentSourceId={contentSourceId}
+              handleContentSource={handleContentSource}
+              contentHosts={contentHosts}
```
Comment (jeremylenz) [inline webpack/scenes/Hosts/ChangeContentSource/index.js:148]: When I change `host_id=` to a bogus ID, I get a `<ContentSourceForm>` but everything is disabled. That case should probably display the error alert instead.

**Review by jeremylenz — COMMENTED** [review]
thanks @stejskalleos!  Some comments below

**Comment by MariSvirik** [comment]
@stejskalleos 
If you are using an alert within a form, you should use an inline alert - view in [PF4 guidelines here](https://www.patternfly.org/v4/components/alert/design-guidelines#using-bordered-inline-alerts).
What's more, after discussion with the content team we came up with this alert to make it super obvious that the source hasn't been changed yet. 

Screenshot 2023-01-13 at 12 46 36

Also, after the button "Update" is clicked and an alert appears, make the button disabled. Otherwise, the "Update" button is a visually primary action even though at this point we want the users to run REX and not repeatedly click on "Update".

There were also, suggestions about changing the button name to "Continue" or "Next" but I would dismiss these suggestions for now.

**Comment by stejskalleos** [comment]
@MariSvirik @jeremylenz PR updated:

- Show alert when host_id is not existing
- Disable Update button after form is submitted
- Alert is `warning` instead of `info`, changed texts
- Bread crumb navigation goes to host or hosts
ccs_monday

File: webpack/scenes/Hosts/ChangeContentSource/components/ContentSourceTemplate.js [hunk 7]
```diff
@@ -39,25 +41,19 @@ const ContentSourceTemplate = ({ template, jobInvocationPath }) => {
...
-                {__('Run job invocation')}
-              </a>
-          &nbsp;
-              {__('to update configuration on all hosts, or')}
-            </>
-          )}
-        &nbsp;
-          {__('update configuration on the hosts manually:')}
-        </p>
+        <Alert variant="warning" title={__('Hosts configurations are not updated yet')} className="margin-top-20" isInline>
```
Comment (jeremylenz) [inline webpack/scenes/Hosts/ChangeContentSource/components/ContentSourceTemplate.js:44]:
```suggestion
        <Alert variant="warning" title={__('Host configurations are not updated yet')} className="margin-top-20" isInline>
```

File: webpack/scenes/Hosts/ChangeContentSource/index.js [hunk 8]
```diff
@@ -74,6 +80,13 @@ const ChangeContentSourcePage = () => {
     }
   };
 
+  const breadcrumbItems = () => {
```
Comment (jeremylenz) [inline webpack/scenes/Hosts/ChangeContentSource/index.js:83]: If it's a single host the breadcrumb should be

```
Hosts > hostname > Change host content source
```

with "Hosts" and "hostname" as links.
If it's multiple hosts the breadcrumb can remain

```
Hosts > Change host content source
```
with "Hosts" as the link.

@MariSvirik can check me, but I think this is how breadcrumbs should be.

File: webpack/scenes/Hosts/ChangeContentSource/index.js [hunk 9]
```diff
@@ -74,6 +80,13 @@ const ChangeContentSourcePage = () => {
...
+    const singleHost = getHostIds(urlParams.host_id).length === 1;
+    const link = singleHost ? { caption: __('Host'), url: foremanUrl(`/new/hosts/${urlParams.host_id[0]}`) } : { caption: __('Hosts'), url: foremanUrl('/hosts') };
```
Comment (jeremylenz) [inline webpack/scenes/Hosts/ChangeContentSource/index.js:85]: When my host id is 28, this link tries to take me to `/new/hosts/2` and not `/new/hosts/28`

**Review by jeremylenz — CHANGES_REQUESTED** [review]
On the CV UI and host detail page, there's a nice banner that shows if no content views are available for the selected environment:

image

---
Would be nice to add that banner here as well.

I would also eventually like to use the same interface overall here (see ChangeHostCVModal.js) but that's probably not in scope here, plus it'll be changing as the multi-CV work progresses.

Some more comments below

**Comment by chris1984** [comment]
Once this is merged, I will start work on Katello 4.7.1

**Comment by stejskalleos** [comment]
@jeremylenz Navigation & texts fixed, for the CV & ENV display I created https://projects.theforeman.org/issues/35958, and added to our backlog

**Comment by jeremylenz** [comment]
Thanks! Sorry, I didn't mention I had already created https://projects.theforeman.org/issues/35559. I went ahead and marked the duplicate.

**Review by jeremylenz — APPROVED** [review]
thanks @stejskalleos!

APJ, just fix the commit message and then I'll merge it.

**Pull request merged**